Sources/ResilientDecoding/Resilient.swift

```diff
@@ -15,163 +15,105 @@ public struct Resilient<Value: Decodable>: Decodable {
-        /// Resilient arrays are reponsible for reporting element decoding errors themselves.
-      } else {
-        resilientDecodingHandled(error)
+  
+  #if DEBUG
+  @dynamicMemberLookup
+  public struct ProjectedValue {
+    public let outcome: ResilientDecodingOutcome
+    
+    public var error: Error? {
+      switch outcome {
+      case .decodedSuccessfully, .keyNotFound, .valueWasNil:
+        return nil
+      case .recoveredFrom(let error, _):
+        return error
       }
-      return Resilient(fallbackValue, error: error)
-    #else
-      resilientDecodingHandled(error)
-      return Resilient(fallbackValue)
-    #endif
+    }
   }
-
+  public var projectedValue: ProjectedValue { ProjectedValue(outcome: outcome) }
+  #endif
+  
 }
 
-// MARK: - Decoding
+// MARK: - Decoding Outcome
 
-struct ResilientDecodingOptions: OptionSet {
-  let rawValue: Int
-  static let suppressKeyNotFoundError = ResilientDecodingOptions(rawValue: 1 << 0)
-  static let suppressValueNotFoundError = ResilientDecodingOptions(rawValue: 1 << 1)
-  static let behaveLikeOptional: ResilientDecodingOptions = [.suppressKeyNotFoundError, .suppressValueNotFoundError]
+#if DEBUG
+public enum ResilientDecodingOutcome {
+  case decodedSuccessfully
+  case keyNotFound
+  case valueWasNil
+  
+  /// https://github.com/apple/swift/blob/88b093e9d77d6201935a2c2fb13f27d961836777/stdlib/public/Darwin/Foundation/JSONEncoder.swift#L1657-L1661
+  case recoveredFrom(Error, wasReported: Bool)
 }
+#else
+struct ResilientDecodingOutcome {
+  static let decodedSuccessfully = Self()
+  static let keyNotFound = Self()
+  static let valueWasNil = Self()
+  static let recoveredFromDebugOnlyError = Self()
+  static func recoveredFrom(_: Error, wasReported: Bool) -> Self { Self() }
+}
+#endif
 
-extension KeyedDecodingContainer {
-
-  /**
-   Resiliently decodes a value for the specified key, using `fallback` if an error is encountered.
-   */
-  func resilientlyDecode<T: Decodable>(
-    _ type: T.Type,
-    forKey key: Key,
-    fallback: @autoclosure () -> T,
-    options: ResilientDecodingOptions) -> Resilient<T>
-  {
-    resilientlyDecode(
-      valueForKey: key,
-      fallback: fallback(),
-      options: options,
-      decode: { Resilient(try T(from: $0)) })
-  }
+// MARK: - Convenience
 
-  /**
-   Resiliently decodes a value for the specified key, using `fallback` if an error is encountered.
-   */
-  func resilientlyDecode<T: Decodable>(
-    _ type: T?.Type,
-    forKey key: Key) -> Resilient<T?>
-  {
-    resilientlyDecode(
-      T?.self,
-      forKey: key,
-      fallback: nil,
-      options: .behaveLikeOptional)
-  }
+extension KeyedDecodingContainer {
 
   /**
    Resiliently decodes a value for the specified key, using `fallback` if an error is encountered.
    This form allows the caller to provide their own `Resilient` with custom errors, which is only used for `ResilientArray` and `ResilientRawRepresentable` optionals that define a `decodingFallback`.
```

Comment:
I wonder if this comment should be updated now that we've removed the other forms of this method?

Reply:
Yup, this comment is now spurious.